nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -90,20 +90,55 @@ public String toString() {
         }
     }
 
+    // validator to ensure that valid EL expressions are used in the directory property
+    static final Validator PATH_WITH_EL_VALIDATOR = new Validator() {
```

Comment:
Wondering if this could be in the standard validators class instead?  There's an  ATTRIBUTE_EXPRESSION_LANGUAGE_VALIDATOR  that's unused but looks very similar, just doesn't have the non empty requirement. Seems like a few tweaks could make it more generic with broad application. Thoughts?